walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,89 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Arrays;
 
+/** Given meeting information and event information, returns the times when the meeting could happen that day
+  */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> requiredAttendees = request.getAttendees();
+    Collection<String> allAttendees = new HashSet<String>(requiredAttendees);
+    allAttendees.addAll(request.getOptionalAttendees());
+    int duration = (int) request.getDuration();
+    
+    Collection<TimeRange> potentialTimesAll = findPotentialTimes(events, allAttendees, duration);
+    Collection<TimeRange> potentialTimesOnlyRequired = findPotentialTimes(events, requiredAttendees, duration);
+    if (potentialTimesAll.isEmpty() && !requiredAttendees.isEmpty()) return potentialTimesOnlyRequired;
```

Comment:
It's generally preferable to always have a `{ block }` after a control flow statement like `if`, because otherwise it's easy to misread what it applies to: I might assume the rest of the line is the condition and jump to the next line.